clients/rospy/src/rospy/topics.py

```diff
@@ -439,6 +458,21 @@ def cleanup_cb_wrapper(s):
             
             return True
 
+    def check(self):
+        new_connections = self.connections[:]
+        removed_connection = False
+
+        for fd in self.connection_poll.error_iter():
+            to_remove = [x for x in new_connections if x.fileno() == fd]
+
+            for x in to_remove:
+                removed_connection = True
+                rospydebug("removing connection to %s, connection error detected"%(x.endpoint_id))
+                self._remove_connection(new_connections, x)
+
+        if removed_connection:
+            self.connections = new_connections
```

Comment:
Doesn't this require to lock `c_lock` since it is potentially modifying the `connections`?